vortex-ffi/src/session.rs

```diff
@@ -66,6 +67,13 @@ impl VortexSession {
     }
 }
 
+impl Drop for VortexSession {
+    fn drop(&mut self) {
```

Comment:
I haven't decided if I like this custom Drop impl. The upside is that we will automatically shut down when the FFI caller shuts down the last session. (Otherwise they effectively have to call this function explicitly, which feels like calling `free` on a resource they don't have ownership or knowledge of). 

On the flip side, we could end up in a situation where we drop the runtime (which can leak resources if there are tasks still running) and then immediately create a new one.